docs/docs/code/bash/README.md

```diff
@@ -0,0 +1,199 @@
+
+```bash
+echo $PIPEDREAM_STEPS | jq .trigger.event
+
+# results in { id: 1, name: "Bulbasaur", type: "plant" }
+```
+
+::: tip
+The period (`.`) in front the `trigger.event` in the example is not a typo. This is to define the starting point for `jq` to start traversing down the JSON in the HTTP response.
+:::
+
+### Sending data downstream to other steps
+
+To share data created, retrieved, transformed or manipulated by a step to others downstream, append it to the `$PIPEDREAM_EXPORTS` variable.
+
+An example speaks a thousand words, so here's one passing data from an API to the bash step.
+
+```bash
+# Retrieve the data from an API and store it in a variable
+DATA=`curl --silent https://pokeapi.co/api/v2/pokemon/charizard`
+
+# Write data to $PIPEDREAM_EXPORTS to share with steps downstream
+EXPORT="key:json=${DATA}"
+echo $EXPORT >> $PIPEDREAM_EXPORTS
+```
+
+::: tip
+Don't worry, the special `key` string in the `EXPORT` will automatically reference the current step's name. 
+
+This way you won't have collisions with multiple bash scripts exporting data. Accessing the data is based off of the step's name, no need to edit the `key` string to try and name it something unique.
+:::
+
+## Using environment variables
+
+You can leverage any [environment variables defined in your Pipedream account](/environment-variables/#environment-variables) in a bash step. This is useful for keeping your secrets out of code as well as keeping them flexible to swap API keys without having to update each step individually.
+
+To access them, just append the `$` in front of the environment variable name.
+
+```bash
+echo $POKEDEX_API_KEY
+```
+
+Or an even more useful example, using the stored environment variable to make an authenticated API request.
+
+```bash
+curl --silent -X POST -h "authorization:Bearer $TWITTER_API_KEY" https://api.twitter.com/2/users/@pipedream/mentions
+```
+
+## Raising exceptions
+
+You may need to stop your step immediately. You can use the normal `exit` function available in bash to quit the step prematurely.
+
+```bash
+  echo "Exiting now!" 1>&2
+  exit 1
+```
+
+This will exit the step and output the error message to `stderr` which will appear in the results of the step in the workflow.
+
+## File storage
+
+If you need to download and store files you can place them in the `/tmp` directory.
+
+### Writing a file to /tmp
+
+For example to download a file to `/tmp` using `curl`
+
+```bash
+# Download the current weather in Cleveland in PNG format
+curl --silent https://wttr.in/Cleveland.png --output /tmp/weather.png
+
+# output the contents of /tmp to confirm the file is there
+ls /tmp
+```
+
+### `/tmp` limitations
+
+The `/tmp` directory can store up to 512 megabytes of storage. Also the storage may be wiped or may not exist between workflow executions.
```

Comment:
Can you create a new var in `config.js` to store the max size of the `/tmp` dir? We reference this in a few places.

Reply:
Actually, we should do a couple more things:

- Move all limits to [this section](https://pipedream.com/docs/limits/#disk) of the limits docs, then refer users there from this section.
- Reference the existing language [here](https://pipedream.com/docs/workflows/steps/code/nodejs/working-with-files/#managing-tmp-across-workflow-runs) re: how `/tmp` files remain across workflow runs.